web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+
+    this.xScaleFocus = d3
+      .scaleBand()
+      .rangeRound([marginFocus.left, elementWidth - marginFocus.right])
+      .padding(0.1);
+    this.yScaleFocus = d3
+      .scaleLinear().range([elementHeight - marginFocus.bottom, marginFocus.top]);
+    this.xAxisFocus = this.focus
+      .append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', `translate(0, ${elementHeight - marginFocus.bottom})`);
+    this.yAxis = this.focus
+      .append('g')
+      .attr('transform', `translate(${marginFocus.left}, 0)`)
+      .style('color', BarChartComponent.COLOR_LIGHT_GRAY);
+
+    this.focus.append("text")
+      .attr("text-anchor", "middle") 
+      .attr("transform", 
+        "translate("+(marginFocus.left/2)+","+
+        ((elementHeight-marginFocus.bottom+marginFocus.top)/2)+")rotate(-90)")
+      .attr("fill", BarChartComponent.COLOR_LIGHT_GRAY)
+      .text("Duration of the push (hours)")
+      .style("font", "12px sans-serif");
+
+    this.xScaleBrush = d3
+      .scaleBand()
+      .rangeRound([marginBrush.left, elementWidth - marginBrush.right])
+      .padding(0.1)
+    this.yScaleBrush = d3.scaleLinear().range([this.heightBrush, 0]);
+    this.xAxisBrush = this.brush.append('g')
+      .attr('transform', `translate(0, ${this.heightBrush})`);
+  }
+
+  /**
+   * This function updates the focus barchart based on the data of the 
+   * dropdown selection.
+   * @param dataSelection: Value selected for the bar chart
+   */
+  public updateChart(dataSelection: number): void {
+    let dataSelected = []
+    if (dataSelection === 0) {
+      dataSelected = this.dataAll;
+    } else {
+      dataSelected = this.dataComplete;
+    }
+    if (!dataSelected) { return; }
+
+    const maxDuration = d3.max(dataSelected, (d: BarChartData) => d.duration);
+    if (!maxDuration) { return; }
+
+    // Upate the xScaleFocus, yScaleFocus, xAxisFocus and yAxisFocus
+    // based on the selected data for the focus chart.
+    this.xScaleFocus.domain(dataSelected.map((d: BarChartData) => d.startTime));
+    this.yScaleFocus.domain([0, maxDuration]);
+    this.yAxis.transition().duration(0)
+      .call(d3.axisLeft(this.yScaleFocus));
+    this.updateXAxisFocus();
+
+    // Update xScaleBrush, yScaleBrush and xAxisBrush based on the 
+    // selected data for the brush chart.
+    this.xScaleBrush.domain(dataSelected.map((d: BarChartData) => d.startTime));
+    this.yScaleBrush.domain([0, maxDuration]);
+    this.xAxisBrush.transition().duration(0)
+    .call(d3.axisBottom(this.xScaleBrush).tickSizeOuter(0))
+    .selectAll("text")
+    .attr("opacity", 0); // Hide the x axis labels of the brush chart.
+
+    // Update the focus chart given the selected data.
+    this.changeFocus(dataSelected);
+
+    // Initial the brush bar chart.
+    const brushBars = this.brush.selectAll('rect')
+      .data(dataSelected);
+
+    brushBars
+      .enter()
+      .append('rect')
+      .attr("x", (d: BarChartData) => this.xScaleBrush(d.startTime))
```

Comment:
```suggestion
      .attr('x', (d: BarChartData) => this.xScaleBrush(d.startTime))
```

Same comment in the other similar places.